File: src/dataset/cpp/beer-song_exercism-chrisnp/claude/ClaudeSonnet4_beer_song_v3.cpp
#include "beer_song.h"
#include <string>

namespace {
    constexpr const char* const VERSE_TEMPLATES[] = {
        "No more bottles of beer on the wall, no more bottles of beer.\n"
        "Go to the store and buy some more, 99 bottles of beer on the wall.\n",
        
        "1 bottle of beer on the wall, 1 bottle of beer.\n"
        "Take it down and pass it around, no more bottles of beer on the wall.\n",
        
        "2 bottles of beer on the wall, 2 bottles of beer.\n"
        "Take one down and pass it around, 1 bottle of beer on the wall.\n"
    };
    
    std::string format_verse(int n) {
        std::string result;
        result.reserve(80);
        result += std::to_string(n);
        result += " bottles of beer on the wall, ";
        result += std::to_string(n);
        result += " bottles of beer.\nTake one down and pass it around, ";
        result += std::to_string(n - 1);
        result += " bottles of beer on the wall.\n";
        return result;
    }
}

std::string beer_song::verse(int n) noexcept {
    if (n <= 2) {
        return VERSE_TEMPLATES[n];
    }
    return format_verse(n);
}

std::string beer_song::sing(int start, int end) noexcept {
    std::string result;
    const size_t estimated_size = (start - end + 1) * 80;
    result.reserve(estimated_size);
    
    for (int n = start; n >= end; --n) {
        if (n != start) {
            result += '\n';
        }
        result += verse(n);
    }
    return result;
}

std::string beer_song::sing(int start) noexcept {
    return sing(start, 0);
}

File: src/dataset/Java/beer-song_exercism-uzilan/claude/ClaudeSonnet4_BeerSong_v3.java
import static java.lang.Character.toUpperCase;
import static java.lang.String.format;

public class BeerSong {

    public String sing(final int from, final int count) {
        StringBuilder result = new StringBuilder();
        
        for (int i = from; i >= from - count + 1; i--) {
            if (i == 0) {
                result.append(format(
                        "%s of beer on the wall, %s of beer.\n" +
                                "Go to the store and buy some more, %s of beer on the wall.\n",
                        capitalize(bottles(i)), bottles(i), bottles(99)));
            } else {
                result.append(format(
                        "%s of beer on the wall, %s of beer.\n" +
                                "Take %s down and pass it around, %s of beer on the wall.\n",
                        bottles(i), bottles(i), one(i), bottles(i - 1)));
            }
            
            if (i > from - count + 1) {
                result.append("\n");
            }
        }
        
        return result.append("\n").toString();
    }

    public String singSong() {
        return sing(99, 100);
    }

    private String bottles(int count) {
        switch (count) {
            case 0:
                return "no more bottles";
            case 1:
                return "1 bottle";
            default:
                return count + " bottles";
        }
    }

    private String capitalize(String string) {
        return toUpperCase(string.charAt(0)) + string.substring(1);
    }

    private String one(int count) {
        return count == 1 ? "it" : "one";
    }
}